### PR TITLE
[sm,resourcemanager] Use allocator for node config objects

### PR DESCRIPTION
**Pull request by mlohvynenko**

### CODE REVIEW

File: include/aos/sm/resourcemanager.hpp [hunk 1]
```diff
@@ -336,9 +336,9 @@ class ResourceManager : public ResourceManagerItf, private NonCopyable {
     StaticString<cNodeTypeLen> mNodeType;
     StaticString<cFilePathLen> mConfigPath;
     Error                      mConfigError {ErrorEnum::eNone};
-    NodeConfig                 mConfig;
```
Comment (al1img) [inline include/aos/sm/resourcemanager.hpp:339]: No need to have this member in static allocator. Keep it as is.

File: include/aos/sm/resourcemanager.hpp [hunk 2]
```diff
@@ -336,9 +336,9 @@ class ResourceManager : public ResourceManagerItf, private NonCopyable {
...
+    UniquePtr<NodeConfig>      mConfig;
 
-    StaticAllocator<sizeof(StaticString<cNodeConfigJSONLen>)> mAllocator;
+    mutable StaticAllocator<sizeof(StaticString<cNodeConfigJSONLen>) + 2 * sizeof(NodeConfig)> mAllocator;
```
Comment (al1img) [inline include/aos/sm/resourcemanager.hpp:341]: Remove mutable and adjust size.

Comment (mlohvynenko) [reply include/aos/sm/resourcemanager.hpp:341]: Mutable is needed to allocate tmp node info object in 
`Error ResourceManager::CheckNodeConfig(const String& version, const String& config) const`

File: include/aos/sm/resourcemanager.hpp [hunk 3]
```diff
@@ -338,7 +338,7 @@ class ResourceManager : public ResourceManagerItf, private NonCopyable {
     Error                      mConfigError {ErrorEnum::eNone};
     NodeConfig                 mConfig;
 
-    StaticAllocator<sizeof(StaticString<cNodeConfigJSONLen>)> mAllocator;
+    mutable StaticAllocator<sizeof(StaticString<cNodeConfigJSONLen>) + sizeof(NodeConfig)> mAllocator;
```
Comment (al1img) [inline include/aos/sm/resourcemanager.hpp:341]: Is it mutable due to `CheckNodeConfig` const method?

**Review by MykolaSolyanko — APPROVED** [review]

**Review by mykola-kobets-epam — APPROVED** [review]

**Review by al1img — APPROVED** [review]

**Pull request merged**